plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -224,13 +226,23 @@ else if (config.getPinGlueClientToCurrentRegion()) {
 
     private static AWSCredentialsProvider getAwsCredentialsProvider(GlueHiveMetastoreConfig config)
     {
+        String sessionIdentifier = "trino-session";
+        if(config.isSessionIdentifier()) {
+            try {
+                UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+                sessionIdentifier = ugi.getUserName();
+            } catch (IOException ex) {
```

Comment:
We can rethrow this as `UncheckedIOException`. It should only happen for Kerberos anyway, so should not be a concern for people using S3. Since people would use the user information for auditing, we wouldn't want to use a default if there is a problem getting it.

Reply:
Removed boolean config parameter and added Runtime exception for session identifier.